test/test.py

```diff
@@ -63,6 +83,97 @@ def test_fmpz():
     assert repr(flint.fmpz(-27)) == "-27"
     assert bool(flint.fmpz(0)) == False
     assert bool(flint.fmpz(1)) == True
+    bigstr = '1' * 100
+    big = flint.fmpz(bigstr)
+    assert big.str() == bigstr
+    assert big.str(condense=10) == '1111111111{...80 digits...}1111111111'
+    raises(lambda: pow(flint.fmpz(2), 2, 3), NotImplementedError)
+    assert flint.fmpz(6).gcd(flint.fmpz(9)) == 3
+    assert flint.fmpz(6).gcd(9) == 3
+    raises(lambda: flint.fmpz(2).gcd('asd'), TypeError)
+    assert flint.fmpz(6).lcm(flint.fmpz(9)) == 18
+    assert flint.fmpz(6).lcm(9) == 18
+    raises(lambda: flint.fmpz(2).lcm('asd'), TypeError)
+    assert flint.fmpz(25).factor() == [(5, 2)]
+    n = flint.fmpz(10**100 + 1)
+    assert n.factor() == [
+        (73, 1), (137, 1), (401, 1), (1201, 1), (1601, 1), (1676321, 1), (5964848081, 1),
+        (129694419029057750551385771184564274499075700947656757821537291527196801, 1)]
+
+    assert n.factor(trial_limit=100) == [
+        (73, 1), (137, 1), (401, 1),
+        (2493516234411471571047384039650897753117456334167082044912715710972543643391271845384040149601, 1)]
+    assert n.factor_smooth() == [
+        (73, 1), (137, 1), (401, 1), (1201, 1), (1601, 1),
+        (1296814508839693536173209832765271992846610925502473758289451540212712414540699659186801, 1)]
+    T, F, VE, OE = True, False, ValueError, OverflowError
+    cases = [
+        # (f, [f(-1), f(0), f(1), f(2), ... f(10)]),
+        (lambda n: flint.fmpz(n).is_prime(),
+            [0, 0, 0, 1, 1, 0, 1, 0, 1, 0, 0, 0]),
+        (lambda n: flint.fmpz(n).is_probable_prime(),
+            [0, 0, 0, 1, 1, 0, 1, 0, 1, 0, 0, 0]),
+        (lambda n: flint.fmpz(n).is_perfect_power(),
+            [T, T, T, F, F, T, F, F, F, T, T, F]),
+        (lambda n: flint.fmpz(n).partitions_p(),
+            [0, 1, 1, 2, 3, 5, 7, 11, 15, 22, 30, 42]),
+        (lambda n: flint.fmpz(n).moebius_mu(),
+            [1, 0, 1, -1, -1, 0, -1, 1, -1, 0, 0, 1]),
+        (lambda n: flint.fmpz.fac_ui(n),
+            [OE, 1, 1, 2, 6, 24, 120, 720, 5040, 40320, 362880, 3628800]),
+        (lambda n: flint.fmpz.primorial_ui(n),
+            [OE, 1, 1, 2, 6, 6, 30, 30, 210, 210, 210, 210, 2310, 2310]),
+        (lambda n: flint.fmpz.fib_ui(n),
+            [OE, 0, 1, 1, 2, 3, 5, 8, 13, 21, 34, 55]),
+        (lambda n: flint.fmpz(n).rising(0),
+            [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]),
+        (lambda n: flint.fmpz(n).rising(1),
+            [-1, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10]),
+        (lambda n: flint.fmpz(n).rising(2),
+            [0, 0, 2, 6, 12, 20, 30, 42, 56, 72, 90, 110]),
+        (lambda n: flint.fmpz.bin_uiui(n, 0),
+            [OE, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]),
```

Comment:
These function signatures are generally a bit inconsistent e.g.:
```python
fmpz(n).func()
fmpz.func(n)
fmpz(n).func(m)
fmpz.func(n, m)
```
Is that something that should be kept for backwards compatibility?

Perhaps if a consistent scheme was decided then aliases could be added with more consistent signatures e.g. `fmpz.rising_factorial(n,m)`.

Does it really make sense to have `_ui` in the name for a Python method?

Reply:
This has to be decided on a case by case basis I think. Sometimes a function f(x,y) has a clear primary variable and then x.f(y) makes sense; binomials and rising factorials arguably fall into this category. In other cases a static method makes more sense.

We could also add a namespace of standalone functions (not requiring method calls), but that is a separate issue.

Agree there is no reason to have ui versions in the Python interface.